walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,50 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
+
+  public final int duration = 0;
+  public final Boolean notInclusive = false;
+  public final Boolean inclusive = true;
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+
+    List<TimeRange> availableEventTimes = new ArrayList<>();
+    events.stream()
+          .sorted((firstRange, secondRange) -> firstRange.getWhen().start() - secondRange.getWhen().start())
+          .forEach(e-> {
+            boolean attendee = !Collections.disjoint(e.getAttendees(), attendees);
+            if (attendee) {
+              availableEventTimes.add(e.getWhen());
+            }
+          });
+
+    availableEventTimes.add(TimeRange.fromStartDuration(TimeRange.END_OF_DAY, duration));
+    List<TimeRange> availableMeetingTimes = new ArrayList<TimeRange>();
+    int meetingStartRequest = TimeRange.START_OF_DAY;
+
+    for (int iterator = 0; iterator < availableEventTimes.size() - 1; iterator++) {
```

Comment:
Nit: `i` is preferred to `iterator`. Also, as was mentioned in the other review, this is an index instead of an iterator.

Reply:
Sorry about that! I'll make sure to fix that.